src/Microsoft.CodeQuality.Analyzers/Core/ApiDesignGuidelines/PropagateCancellationTokensWhenPossible.cs

```diff
@@ -33,11 +38,96 @@ public abstract class PropagateCancellationTokensWhenPossibleAnalyzer : Diagnost
 
         public override void Initialize(AnalysisContext analysisContext)
         {
-            // TODO: Enable concurrent execution of analyzer actions.
-            //analysisContext.EnableConcurrentExecution();
+            analysisContext.EnableConcurrentExecution();
 
-            // TODO: Configure generated code analysis.
-            //analysisContext.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.None);
+            // TODO: Include this?
+            // analysisContext.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.None);
+
+            analysisContext.RegisterOperationAction(AnalyzeInvocation, OperationKind.Invocation);
+        }
+
+        private static void AnalyzeInvocation(OperationAnalysisContext context)
+        {
+            var invocation = (IInvocationOperation)context.Operation;
+            if (InvocationShouldPassCancellationToken(invocation, context.Compilation))
+            {
+                context.ReportDiagnostic(invocation.Syntax.CreateDiagnostic(Rule));
+            }
+        }
+
+        private static bool InvocationShouldPassCancellationToken(IInvocationOperation invocation, Compilation compilation)
+        {
+            var cancellationTokenType = WellKnownTypes.CancellationToken(compilation);
+
+            var targetMethod = invocation.TargetMethod;
+            if (targetMethod == null)
+            {
+                return false;
+            }
+
+            if (!AcceptsCancellationTokens(targetMethod))
+            {
+                // Check if there is a different overload of the method that accepts a CancellationToken.
+                var methodName = targetMethod.Name;
+                var overloads = targetMethod.ContainingType.GetMembers(methodName).OfType<IMethodSymbol>();
+                var targetOverload = overloads.FirstOrDefault(
+                    m => AcceptsCancellationTokens(m) && StartsWithSameParameters(m, targetMethod));
```

Comment:
Self-note: Check that `m` is exactly the last parameter instead of any parameter, and that `m.Parameters.Length == targetMethod.Parameters.Length + 1`